website/docs/r/namespace.html.markdown

```diff
@@ -0,0 +1,32 @@
+---
+layout: "nomad"
+page_title: "Nomad: nomad_namespace"
+sidebar_current: "docs-nomad-resource-namespace"
+description: |-
+  Provisions a namespace within a Nomad cluster.
+---
+
+# nomad_namespace
+
+Provisions a namespace within a Nomad cluster.
+
+~> **Enterprise Only!** This API endpoint and functionality only exists in
+Nomad Enterprise. This is not present in the open source version of Nomad.
+
+## Example Usage
+
+Registering a namespace:
+
+```hcl
+resource "nomad_namespace" "dev" {
+  name = "dev"
+  description = "Shared development environment."
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+- `name` `(string: <required>)` - A unique name for the namespace.
+- `description` `(string: "")` - A description of the namespace.
```

Comment:
I might be wrong, but isn't the convention:

```
* `name` - (Required) A unique name for the namespace. Changing this forces a new resource to be created.
* `description` - (Optional) Description for this namespace.
* `quota` - (Optional) Quota for this namespace.
```

(note: quota is missing above)

Reply:
I think in general, yes, but when I got to the provider, it already had the above format, so I was trying to keep it consistent within the provider :/ Happy to do a sweep to update, though I have to admit I _do_ think the version I copied is a bit more informative, providing information on defaults and types.

Reply:
Oh, also, quota is intentionally missing until #25 gets merged, because I have no way to test it until then. I just included the code here because I was already in the code, but I don't want to document it until I can actually test it.